src/main/java/codeu/model/store/persistence/PersistentDataStore.java

```diff
@@ -170,27 +170,28 @@ public PersistentDataStore() {
 
     List<Mention> mentions = new ArrayList<>();
 
-    // Retrieve all messages from the datastore.
+    // Retrieve all mentions from the datastore.
     Query query = new Query("chat-mentions");
   
-      for (Entity entity : results.asIterable()) {
-          try {
-
-            Set<String> dataStoreMessageIds = new HashSet<>((Collection<String>)
+    PreparedQuery results = datastore.prepare(query);
+  
+    for (Entity entity : results.asIterable()) {
+      try {
+        Set<String> dataStoreMessageIds = new HashSet<>((Collection<String>)
```

Comment:
Is the datastore reading/writing the Sets? I was also having compilation problems and it was related to trying to read/write other Collections like ArrayList

Reply:
It is reading from datastore but the problem previously is that we made a query and never ran it. So it was never getting any results and was not running this for loop For the error you are getting is this when you pulled from master or before?